src/decoder.rs

```diff
@@ -179,22 +124,49 @@ fn decode_c_reg_based(raw: usize) -> Instr {
         0b111 => {
             let rs2 = rd_rs2;
             let imm = 0b000 | (raw >> 7) & 0b111000 | ((raw << 1) & 0b11000000);
-            Instr::CSd { rs1, rs2, imm }
+
+            Instr::Store {
+                rs2,
+                rs1,
+                imm: (imm * 8).try_into().unwrap(),
+                len: (64),
+                compressed: (true),
```

Comment:
Not really important, but any reasons for the parenthesis here?

Reply:
Not really, my IDE put them in place as a suggestion and `just test` didn't deem them as unnecessary.
I can remove them, np :)